app/src/main/java/com/github/sdp/ratemyepfl/viewmodel/UserProfileViewModel.kt

```diff
@@ -34,7 +34,14 @@ class UserProfileViewModel @Inject constructor(
     }
 
     fun addClass(c: Class) {
-        TODO()
+        if (currentUser.isLoggedIn()) {
+            viewModelScope.launch {
+                val user = currentUser.getUserId()?.let { userDatabase.getUserByUid(it) }
```

Comment:
I think you can omit this line, you already have the id of the user.

Reply:
yes, nice catch ! will change it next sprint